### PR TITLE
Create Calculator Console App

### PR DESCRIPTION
**Pull request by mustaqeemtariq**
Task is attached for reference:
https://dev.azure.com/CodeNinjaConsulting/Talent%20Pool/_workitems/edit/1410

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+﻿using System;
+class Program
+{
+    static void Main()
+    {
+        Console.WriteLine("--------------------------\n\n Welcome to Calculator \n\n--------------------------\n");
```
Comment (hassan475) [inline Program.cs]: Please create a static class like "ApplicationMessages" that will contain all of your static read-only strings.

File: Program.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        while (true)
+        {
+            Console.WriteLine("Select one of the following operations\n1.Add\n2.Subtract\n3.Multiply\n4.Divide");
+            string input = Console.ReadLine();
+            if (String.IsNullOrEmpty(input))
+            {
+                Console.WriteLine("Input is required");
+                continue;
+            }
+            if (input == "1" || input == "2" || input == "3" || input == "4")
```
Comment (hassan475) [inline Program.cs]: Avoid using "Magic numbers", Try with Enums in C#.

**Review by hassan475 — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 1 above)
Comment (mustaqeemtariq) [reply Program.cs]: Done.

File: Program.cs (hunk 2 above)
Comment (mustaqeemtariq) [reply Program.cs]: Done.

File: Messages.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+public static class ApplicationMessage
+{
+    public static void ShowWelcomeMessage()
+    {
+        Console.WriteLine("--------------------------\n\n Welcome to Calculator \n\n--------------------------\n");
+    }
+
+    public static void ShowExitMessage()
+    {
```
Comment (hassan475) [inline Messages.cs]: create your static class like this:
` public static class ApplicationMessage
    {
        /// <summary>
        /// The CSI Password
        /// </summary>
        public static const string PasswordCsi = "CsystemsIdeal1";
}`

**Review by hassan475 — CHANGES_REQUESTED** [review]

File: Messages.cs (hunk 3 above)
Comment (mustaqeemtariq) [reply Messages.cs]: Done

File: Messages.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+public static class ApplicationMessage
+{
+    static readonly string welcomeMessage = "--------------------------\n\n Welcome to Calculator \n\n--------------------------\n";
+    static readonly string exitMessage = "Goodbye!";
+    static readonly string menuOptionMessage = "Select one of the following operations\n1.Add\n2.Subtract\n3.Multiply\n4.Divide";
+    static readonly string nullMessage = "Input is required";
+    static readonly string validInputMessage = "Invalid input - must be a valid integer";
+    static readonly string validOptionMessage = "Please enter a valid menu option.\n";
+    static readonly string confirmExitMessage = "Do you want to perform another calculation. Press No to exit program: ";
+    public static void ShowWelcomeMessage()
```
Comment (hassan475) [inline Messages.cs]: You can avoid these static functions, simply call static strings directly.

**Review by hassan475 — CHANGES_REQUESTED** [review]

File: Messages.cs (hunk 4 above)
Comment (mustaqeemtariq) [reply Messages.cs]: Updated

File: Program.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+}
+class Program
+{
+    static void Main()
+    {
+        Console.WriteLine(ApplicationMessage.welcomeMessage);
+        while (true)
+        {
+            MenuOption option = GetMenuOption();
+            int num1 = GetIntegerInput("Enter the first number: ");
```
Comment (hassan475) [inline Program.cs]: Move these messages to static class as well.

**Review by hassan475 — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
+﻿using System;
+
+enum MenuOption
```
Comment (hassan475) [inline Program.cs]: Move this enum code to a separate cs file.

**Review by hassan475 — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 6 above)
Comment (mustaqeemtariq) [reply Program.cs]: Done

File: Program.cs (hunk 5 above)
Comment (mustaqeemtariq) [reply Program.cs]: Done

**Review by hassan475 — APPROVED** [review]
Looking good.

**Pull request merged**